Generic non-ref type conversion causes error at C compiler

``` nimrod
type
  fooObj[T] = object of RootObj
  bazObj[T] = object of fooObj[T]
    x: T

var troz: fooObj[string]
echo bazObj[string](troz).x
```

Fails with:

```
nimcache/x.c: In function ‘xInit’:
nimcache/x.c:144:2: error: conversion to non-scalar type requested
  printf("%s\012", ((bazobj88034) (troz_88025)).x? (((bazobj88034) (troz_88025)).x)->data:"nil");
  ^
nimcache/x.c:144:2: error: conversion to non-scalar type requested
```

At least it should fail with an error by the Nim compiler.

--- a/tests/template/template_issues.nim
+++ b/tests/template/template_issues.nim
@@ -271,6 +271,6 @@ ggg(hello)
 
 
 # bug #2586
-var x = 10'u8
-echo x < 9 # Works
-echo x > 9 # Error: type mismatch
+var z = 10'u8
+echo z < 9 # Works
+echo z > 9 # Error: type mismatch

--- a/tests/template/template_issues.nim
+++ b/tests/template/template_issues.nim
@@ -11,6 +11,8 @@ hey
 foo
 foo
 foo
+false
+true
 '''
 """
 
@@ -266,3 +268,9 @@ iterator hello() : int {.closure.} =
     
 foof(int)
 ggg(hello)
+
+
+# bug #2586
+var x = 10'u8
+echo x < 9 # Works
+echo x > 9 # Error: type mismatch

--- a/tissues.nim
+++ b/tissues.nim
@@ -16,3 +16,13 @@ f(nil)
 type A = object #of RootObj <-- Uncomment this to get no errors
   test: proc(i: A): bool
 var a: proc(i: A): bool # Or comment this line to get no errors
+
+
+# bug #2703
+type
+  fooObj[T] = object of RootObj
+  bazObj[T] = object of fooObj[T]
+    x: T
+
+var troz: fooObj[string]
+echo bazObj[string](troz).x

